### PR TITLE
:sparkles: add ability to pass maven coordinate in location

### LINKED ISSUE #554: Add support for taking maven artifact as input so addon doesnt have to deal with downloading artifacts
Analyzer addon currently takes artifact coordinates as input and downloads them prior to running analysis, this is a provider specific logic that exists in the addon and it would be better for the addon to be agnostic of specifics of the provider. The ask is that java provider should download the artifacts itself.

### PR DESCRIPTION
**Pull request by pranavgaikwad**
Fixes #554

### CODE REVIEW

File: docs/providers.md [hunk 1]
```diff
@@ -92,7 +92,7 @@ Here's an example config for `java` provider that is currently in-tree and does
 }
 ```
 
-The `location` can be a path to the application's source code or to a binary JAR, WAR, or EAR file.
+The `location` can be a path to the application's source code or to a binary JAR, WAR, or EAR file. Optionally, coordinates to a maven artifact can be provided as input in the format `mvn://<group-id>.<artifact-id>:<version>:<classifier>:<path>`. The field `<path>` is optional, it specifies a local path where the artifact will be downloaded. If not specified, provider will use a temporary directory to download it.
```
Comment (shawn-hurley) [inline docs/providers.md:95]: can we use something like `@` or something that is not a part of a common URI for the path part?

**Review by shawn-hurley — APPROVED** [review]

File: docs/providers.md (hunk 1 above)
Comment (shawn-hurley) [reply docs/providers.md:95]: This is a minor concern

Comment (pranavgaikwad) [reply docs/providers.md:95]: sure can

File: external-providers/java-external-provider/pkg/java_external_provider/provider.go [hunk 2]
```diff
@@ -220,6 +221,42 @@ func (p *javaProvider) Init(ctx context.Context, log logr.Logger, config provide
 	var returnErr error
 	// each service client should have their own context
 	ctx, cancelFunc := context.WithCancel(ctx)
+	// location can be a coordinate to a remote mvn artifact
+	if strings.HasPrefix(config.Location, MvnURIPrefix) {
+		mvnUri := strings.Replace(config.Location, MvnURIPrefix, "", 1)
+		// URI format is <group>:<artifact>:<version>:<classifier>:<path>
+		// <path> is optional & points to a local path where it will be downloaded
```
[the remark this answers is not in the document]
Comment (pranavgaikwad) [reply external-providers/java-external-provider/pkg/java_external_provider/provider.go:228]: @jmle yeah, in the hub UI, we have four fields (defaulting classifier to JAR)

[the remark this answers is not in the document]
Comment (pranavgaikwad) [reply external-providers/java-external-provider/pkg/java_external_provider/provider.go:228]: @jmle done, defaulting to jar now

**Review by shawn-hurley — APPROVED** [review]

**Pull request merged**